### PR TITLE
fix: filling users logic

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes https://app.campsite.com/cal/posts/lojq7cf8uuk1

### Tested

Screenshot 2024-12-11 at 10 11 41 PM

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Login with Admin account
2. Create a team and team event-type
3. Enable this setting

Screenshot 2024-12-11 at 10 14 02 PM

4. Go to booking page. You should be seeing something like:

Screenshot 2024-12-11 at 10 14 15 PM

5. Test the OG image

### CODE REVIEW

File: apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx [hunk 1]
```diff
@@ -103,21 +111,46 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
...
+        },
+      },
+    });
+    users = data
+      .filter((user) => user.username)
+      .map((user) => ({
+        username: user.username ?? "",
+        name: user.name ?? "",
+      }));
+  }
```
Comment (hbjORbj) [inline apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx:150]: i wrote this minimally based on what we currently do in getPublicEvent

File: apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx [hunk 2]
```diff
@@ -169,11 +207,7 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
           username: orgSlug ?? null,
         },
         title: eventData.title,
-        users: usersInOrgContext.map((user) => ({
-          ...user,
-          metadata: undefined,
-          bookerUrl: getBookerBaseUrlSync(user.profile?.organization?.slug ?? null),
-        })),
+        users,
```
Comment (Udit-takkar) [inline apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx:210]: This doesn't contain `bookerUrl`. is this backward compatible?

Comment (hbjORbj) [reply apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx:210]: We only recently passed users and we only use username and name from users. check out https://github.com/calcom/cal.com/pull/17987 if you need

**Review by emrysal — APPROVED** [review]
Looks good to me

**Review by zomars — APPROVED** [review]

**Pull request merged**